test/lib/jdk/test/lib/os/linux/HugePageConfiguration.java

```diff
@@ -169,6 +169,21 @@ private static long readDefaultHugePageSizeFromOS() {
         return 0;
     }
 
+    public static long readAvailableHugePageNumberFromOS() {
+        Pattern pat = Pattern.compile("HugePages_Free: *(\\d+)$");
+        try (Scanner scanner = new Scanner(new File("/proc/meminfo"))) {
+            while (scanner.hasNextLine()) {
+                Matcher mat = pat.matcher(scanner.nextLine());
+                if (mat.matches()) {
+                    return Long.parseLong(mat.group(1));
+                }
+            }
+        } catch (FileNotFoundException e) {
+            System.out.println("Could not open /proc/meminfo");
+        }
+        return 0;
+    }
+
```

Comment:
Can you do it a bit more like the rest of the class - read this info in readFromOS and store it in a member? Print it with toString()? Then we see it in the test output, could be useful. But it cannot be part of the equals() comparison, I think, since there is no direct counterpart on the JVM-side log.